[dogfood] Worktree CLI binary crashes with ERR_MODULE_NOT_FOUND for @red-codes/storage, silently breaking governance hooks

## Governance Issue

**Agent:** workspace-pr-review-agent (claude-code:opus:reviewer)
**Version:** agentguard v2.10.3 (main worktree) / worktree binary crashes before version
**Repo:** AgentGuardHQ/agentguard-workspace
**Box:** $(hostname) / $(whoami)
**Action attempted:** Starting governance runtime via `node apps/cli/dist/bin.js claude-hook pre --store sqlite` (executed by `scripts/claude-hook-wrapper.sh` as PreToolUse hook)
**Expected behavior:** Governance hook intercepts tool calls and enforces policy/invariants
**Actual behavior:** Node.js crashes with `ERR_MODULE_NOT_FOUND: Cannot find package '@red-codes/storage'` — the worktree CLI binary can't resolve workspace packages at runtime

### Root Cause

When a scheduled agent worktree is created for `workspace-pr-review-agent-2858546`, the startup hook runs `pnpm build` which partially fails:
- Packages in the shared workspace (`@red-codes/core`, `@red-codes/storage`, etc.) all report `cache hit` and succeed
- The CLI app (`@red-codes/agentguard`) has a `cache miss` and fails with 100+ `TS2307: Cannot find module '@red-codes/core'` errors

The `apps/cli/dist/bin.js` binary exists (from a previous build) but when executed from the **worktree path**, Node.js ESM resolution fails to find `@red-codes/storage` because the worktree's `node_modules` symlinks don't resolve correctly to the shared workspace packages.

### Impact

- `scripts/claude-hook-wrapper.sh` finds the binary at `$AGENTGUARD_WORKSPACE/apps/cli/dist/bin.js` and attempts to execute it
- The binary crashes with ERR_MODULE_NOT_FOUND before governance evaluation occurs
- Governance is **silently broken** for all tool calls in the worktree — fail-open or hook error state
- The worktree is for a **reviewer agent** (read-only) so no destructive actions were possible, but any development agent worktree with this issue would have ungoverned write access

### Policy (agentguard.yaml)
```yaml
$(cat agentguard.yaml 2>/dev/null | head -30 || echo '(not found)')
```

### Persona
```
AGENTGUARD_PERSONA_DRIVER=claude-code
AGENTGUARD_PERSONA_MODEL=unknown
AGENTGUARD_PERSONA_ROLE=reviewer
AGENTGUARD_PERSONA_PROJECT=workspace-pr-review-agent-2858546
AGENTGUARD_PERSONA_TRUST_TIER=standard
```

### Hook settings
<details><summary>.claude/settings.json</summary>

```json
$(cat .claude/settings.json 2>/dev/null || echo '(not found)')
```
</details>

### Relevant log output
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@red-codes/storage' imported from /home/jared/agentguard-workspace/.worktrees/workspace-pr-review-agent-2858546/apps/cli/dist/bin.js
    at Object.getPackageJSONURL (node:internal/modules/package_json_reader:314:9)
    at packageResolve (node:internal/modules/esm/resolve:768:81)
    at moduleResolve (node:internal/modules/esm/resolve:855:18)
    at defaultResolve (node:internal/modules/esm/resolve:985:11)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:731:20)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:708:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:310:38)
    at ModuleJob._link (node:internal/modules/esm/module_job:182:49) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Build failure context from session startup (abbreviated):
```
@red-codes/agentguard:build: cache miss, executing 33b4abb4a0390dbf
@red-codes/agentguard:build: ../../packages/events/src/schema.ts(12,8): error TS2307: Cannot find module '@red-codes/core'
... (100+ similar errors across all packages)
@red-codes/agentguard:build: ELIFECYCLE  Command failed with exit code 2.
```

### Suggested Fix

The `start-governance-runtime` skill and/or `scripts/claude-hook-wrapper.sh` should fall back to the **main worktree binary** when the local worktree binary fails at startup. The main worktree always has a working binary. Alternative: the worktree `pnpm build` should succeed reliably so the local binary is always valid.

A potential workaround in `claude-hook-wrapper.sh`:
```bash
# Try local binary first, fall back to main worktree if it crashes
if ! node "$AGENTGUARD_WORKSPACE/apps/cli/dist/bin.js" --version &>/dev/null; then
  MAIN_ROOT=$(git worktree list --porcelain | head -1 | sed 's/worktree //')
  if [ -f "$MAIN_ROOT/apps/cli/dist/bin.js" ]; then
    AGENTGUARD_BIN="node $MAIN_ROOT/apps/cli/dist/bin.js"
  fi
fi
```

---
*Filed by workspace-pr-review-agent during scheduled run 2026-03-29*

--- a/scripts/claude-hook-wrapper.sh
+++ b/scripts/claude-hook-wrapper.sh
@@ -23,6 +23,20 @@ elif command -v agentguard &>/dev/null; then
   AGENTGUARD_BIN="agentguard"
 fi
 
+# Probe the local binary — in git worktrees, apps/cli/dist/bin.js may exist but
+# crash with ERR_MODULE_NOT_FOUND because node_modules is only in the main workspace.
+# (Node ESM resolves packages relative to the binary path; worktrees don't have
+# their own node_modules.) Fall back to the main worktree binary if probe fails (#1376).
+if [ -n "$AGENTGUARD_BIN" ] && ! $AGENTGUARD_BIN --version >/dev/null 2>&1; then
+  _MAIN_ROOT="$(git worktree list --porcelain 2>/dev/null | sed -n '1s/^worktree //p')"
+  if [ -n "$_MAIN_ROOT" ] && [ -f "$_MAIN_ROOT/apps/cli/dist/bin.js" ]; then
+    AGENTGUARD_BIN="node $_MAIN_ROOT/apps/cli/dist/bin.js"
+  else
+    # No fallback available — clear so bootstrap exemption handles the action
+    AGENTGUARD_BIN=""
+  fi
+fi
+
 # BOOTSTRAP EXEMPTION (AgentGuardHQ/agentguard#995):
 # When the kernel binary is missing, allow bootstrap commands (install/build)
 # and read-only tools through so the agent can self-bootstrap.

